### PR TITLE
added cart functionality

### PR DESCRIPTION
**Pull request by decode9**

### CODE REVIEW

**Comment by the-code-workers** [comment]
Your Code failure

**Comment by the-code-workers** [comment]
Your Code is succesfully Success

File: src/store/cart/saga.ts [hunk 1]
```diff
@@ -1,11 +1,99 @@
...
-  yield call(showDialog, 'Producto agregado al carrito')
+function* addedItemAsync({ payload }: any) {
+  try {
+    const { user: { user: { sessionToken }, isAuth } } = yield select(state => state)
+
+    yield put(actionObject(SHOW_LOADER, true))
+
+    const { item, quantity, variation } = payload
+
+    console.log(variation)
```
Comment (rotcax) [inline src/store/cart/saga.ts]: Please remove this console log

File: src/store/cart/saga.ts [hunk 2]
```diff
@@ -1,11 +1,99 @@
...
+import { SIGN_UP_ASYNC } from '@store/user/action-types'
+import { actionObject, filter, GraphQlClient, manageError, reduceVariation, showDialog, validateFetch } from '@utils'
+import { put, select } from 'redux-saga/effects'
+import { ADDED_ITEM, REMOVE_ITEM, SET_ITEM, UPDATE_QUANTITY } from './action-types'
 
-function* addedItemAsync() {
-  yield call(showDialog, 'Producto agregado al carrito')
+function* addedItemAsync({ payload }: any) {
+  try {
+    const { user: { user: { sessionToken }, isAuth } } = yield select(state => state)
```
Comment (rotcax) [inline src/store/cart/saga.ts]: For this case, create a selector, for example:

In selectors.ts: export const getUser = (state: any) => state.user
In cart saga: const { user: { user: { sessionToken }, isAuth } } = yield select(getUser)

File: src/store/cart/saga.ts [hunk 3]
```diff
@@ -1,11 +1,99 @@
...
+  }
+}
+
+
+function* updateQuantityAsync({ payload: { product, type } }: any) {
+  try {
+
+    yield put(actionObject(SHOW_LOADER, true))
+
+    const { user: { user: { sessionToken } } } = yield select(state => state)
```
Comment (rotcax) [inline src/store/cart/saga.ts]: In this case also use a selector

File: src/store/page/saga.ts [hunk 4]
```diff
@@ -29,6 +31,7 @@ const getPageByName = (name) => {
   const query = `
     query Page {
       ${pages[name]}
+      ${cartQuery}
```
Comment (rotcax) [inline src/store/page/saga.ts:35]: If required cart query in all pages for the website?

File: src/store/user/saga.ts [hunk 5]
```diff
@@ -25,15 +25,17 @@ function* signUpAsync({ payload }: any) {
...
     yield put(actionObject(SHOW_LOADER, true))
 
     let response = yield call(GraphQlClient, loginUser(payload))
     response = validateFetch(response)
 
     yield put(actionObject(SIGN_UP_ASYNC, {
       user: {
-        ...response.login.user,
-        authToken: response.login.authToken
+        ...{ ...response.login.user, ...response.login.customer },
```
Comment (rotcax) [inline src/store/user/saga.ts]: This is equal to ...response.login.user, ...response.login.customer.

Use ...{ ...rest } is unnecessary

**Review by rotcax — CHANGES_REQUESTED** [review]

**Comment by the-code-workers** [comment]
Your Code is succesfully Success

File: src/store/page/saga.ts (hunk 4 above)
Comment (decode9) [reply src/store/page/saga.ts:35]: This configuration was made for have cart data updated in page load

**Review by rotcax — APPROVED** [review]

**Pull request merged**